core/src/main/java/inetsoft/graph/internal/GDebug.java

```diff
@@ -116,7 +116,10 @@ public static void writeImage(String fname, Image img) {
     */
    public static void writeImage(File file, Image img) {
       try(java.io.FileOutputStream fout = new java.io.FileOutputStream(file)) {
-         CoreTool.writePNG(img, fout);
+         //write to in-memory buffer first to prevent issues with Spring async response output stream
```

Comment:
This is unnecessary. It only affects the Tomcat async servlet request's output stream. It should not be a general problem.